### PR TITLE
cataclysm-dda: split off 0.D release branch

### PR DESCRIPTION
**Pull request by l29ah**

### CODE REVIEW

**Comment by leycec** [comment]
**Awesome.** Admittedly, it took me a few minutes to grok the *exact* intention of this pull request. Assuming you submit future pull requests (...*please do!*), it'd be useful if you could summarize your changeset with human-readable description. You know – just sayin'.

Also, I'm unconvinced that I want to permanently maintain two separate Cataclysm: DDA ebuilds as this changes suggests: e.g., both `cataclysm-dda-9999.0d.ebuild` *and* `cataclysm-dda-9999.9999.ebuild`. I'm concerned that end users will fail to understand the obscure versioning conventions we've adopted and then endlessly ridicule me about my awful design choices on our issue tracker.

That said, I love your specific changes to `cataclysm-dda-9999.0d.ebuild` and admit that most end users probably would prefer to reside on the `0.D` branch – especially as that branch *finally* appears to be stabilizing.

In other words, there's quite a bit for me to meaningfully chew on here. I'll probably tackle the unrelated C:DDA issue #69 first, which is both the low-hanging fruit and shameful. Unfortunately, doing so will probably break this pull request. Fortunately, given the above concerns, I was considering manually integrating only a portion of this changeset anyway.

In short, I have no idea what I'm going to do here. It saddens me that the C:DDA codebase has become so bifurcated. They *really* fell into the developer hell hole, didn't they?

**Comment by l29ah** [comment]
Yes, symlinking would do the job, but the main Gentoo tree doesn't use symlinks, so i suppose they don't work for some Gentoo systems.

**Comment by leycec** [comment]
> the main Gentoo tree doesn't use symlinks

**Yup.** It probably should, frankly. Duplication is the death of sanity.

> so i suppose they don't work for some Gentoo systems.

...doubtful, but possibly possible. All POSIX-compliant platforms – which is all platforms compatible with Gentoo Prefix – support symlinking. Given that this and numerous other overlays have leveraged ebuild symlinks to reduce DRY for over two decades with no appreciable issues, my lurking suspicion is that there are no rational reasons for Portage *not* to do so.

Fortunately, we ain't Portage. :wink: 

Thanks again for your stupendous contribution. You're an excellent human specimen, Sergey! I apologize in advance if I break this changeset by resolving #69 – and promise to patch things back up again. We'll get these welcome improvements integrated in a jiffy. <sup>...*where "jiffy" is defined as "in a week or three."*</sup>

**Comment by leycec** [comment]
**Boom!** :boom: 

At long last, #69 has now been resolved. Although this pull request superficially appears to have no conflicts, in actuality pretty much everything has changed and will need to be rethought a bit. Of course, that's *my* responsibility. `</sigh>`

I've sat on this depressingly long enough. Now, it's time to see what I can salvage from your most excellent work. I may not necessarily branch the live `9999` ebuild into two separate `9999` ebuilds as you'd prefer. Now that 0.D has been officially released, all development work appears to be confined to the `master` branch.

However, I will endeavour to manually merge everything else of value into the existing `9999` ebuild – which is quite a lot, actually. Let's see where this dark road leads...

**Comment by leycec** [comment]
**O.K.!** [Commit 23ff19b922](https://github.com/leycec/raiagent/pull/70/commits/23ff19b922cf9f2c06f7d65b6146d00102346ecf) is an *excellent* catch. You're absolutely right. Allowing `BACKTRACE=0` would be highly undesirable under Gentoo (and most platforms, frankly), as doing so both strips binaries *and* entirely disables crash reporting. Luckily, `BACKTRACE=1` incurs no performance penalties.

For sanity, I've manually merged the contents of that commit into 6e1317829195d with attribution to you. Thanks for that, @l29ah. You've made the post-apocalyptic world a better place for disgustingly disfigured mutants.

Sadly, the entirety of [commit 255d5c53859](https://github.com/leycec/raiagent/pull/70/commits/255d5c53859b4d23841252bfc469f7a3f98ee099) no longer appears to apply. The 0.D branch is [**2216 commits** behind `master`](https://github.com/CleverRaven/Cataclysm-DDA/tree/0.D-branch) and is no longer actively maintained. Since the HEAD of that branch is identical (or at least closely comparable) to the stable 0.D release tarball, there's not much point in explicitly tracking that branch. Unless I'm missing something really obvious? <sup>...*I'm probably missing something really obvious.*</sup>

With grim apologies, I'm reluctantly closing the remainder of this pull request unmerged. You clearly injected a non-trivial syringe full of blood, sweat, and tears into this contribution. I *hate* rejecting your generous work.

But don't feel bad! It's basically *my* fault. If I hadn't taken so long to resolve #69, [commit 255d5c53859](https://github.com/leycec/raiagent/pull/70/commits/255d5c53859b4d23841252bfc469f7a3f98ee099) probably would have been the Right Thing to Do™. I hope this doesn't dissuade you from submitting any future changes. The `BACKTRACE=1` catch was excellent – and something I couldn't have caught on my own.

I too now feel like a complete jerk.
